client/components/PubHeader/PubHeader.js

```diff
@@ -67,10 +65,6 @@ class PubHeader extends Component {
 		}
 	}
 
-	handleTitleChange(newTitle) {
-		this.setState({ title: newTitle.replace(/\n/g, '') });
```

Comment:
Hrm...that doesn't feel like a great reason to use state, and I think the only alternative to uncontrolled is memoization, which seems like a bad idea.

What about doing the replacement when we send it to the server? The downside is it'll allow the user to do something in the frontend that we don't actually support, but, a, it feels like an edge case, and it will be corrected as soon as they hit enter and the title makes the round-trip.

Reply:
I think we'll need to use [getDerivedStateFromProps](https://github.com/pubpub/pubpub/pull/303/files#r249204024) here. And in there, have something like
```
constructor(props) {
   super(props);
   this.state = {
      title: props.pubData.title,
      prevPropsTitle: props.pubData.title;
   };
}
static getDerivedStateFromProps(props, state) {
   if (state.previousPropsTitle !== props.pubData.title) {
      return {
         title: props.pubData.title,
         prevPropsTitle: props.pubData.title,
      };
   }
   return null;
}
```

Reply:
I looked at that, but it feels icky. Let me try one more thing before we go there.